python/pyspark/sql/pandas/group_ops.py

```diff
@@ -503,7 +503,11 @@ def transformWithStateUDF(
             statefulProcessorApiClient.set_implicit_key(key)
             result = statefulProcessor.handleInputRows(key, inputRows)
 
-            return result
+            try:
+                yield result
+            finally:
+                statefulProcessor.close()
```

Comment:
Shall we set handle state to `CLOSE` here?

Reply:
I just realized an issue that this is actually being called after processing each grouping key instead of finishing processing all keys for a microbatch. I'll need to revisit this to see if there's a good way to handle this (I cannot think about a good way to detect `if the current key is the last key to process` right now), if it's not a quick fix, we can probably exclude it for now and have a followup PR fixing it. cc @HeartSaVioR

Reply:
Maybe we could try injecting a dummy row at the end of the iterator in `writeNextInputToArrowStream` indicating all the keys have been processed, but I'll need to do some experiments first.

Reply:
I don't feel like current interface would give you such information - we'll probably need to have another control message to send the signal from JVM to Python (UDF). I agree this may take time, but probably need to mark it as a blocker so that we address before the release.